### PR TITLE
Don't crash if you reference a generator built against .NET Framework

### PR DESCRIPTION
**Pull request by jasonmalinowski**
The compiler added a block to prevent loading generators that are built against the .NET Framework, since we know long term that we'll want them to be runnable on .NET Core. The IDE helper that deals with converting the error code to an IDE diagnostic needs to be updated to support this new error, or otherwise Visual Studio will crash.

Fixes https://github.com/dotnet/roslyn/issues/47845

### CODE REVIEW

File: src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs
Comment (jasonmalinowski) [inline src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs:45]: @mavasani Is just picking the next AD... code like this is OK or does other stuff need to be updated to?

File: src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs
Comment (jasonmalinowski) [inline src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs:206]: @chsienki -- what's the error code for the C# one? It looks like no code was ever added to the command line compiler as far as I can tell?

File: src/Compilers/CSharp/Portable/CSharpResources.resx
Comment (jasonmalinowski) [inline src/Compilers/CSharp/Portable/CSharpResources.resx:6569]: XML formatting got a bit screwed up here.

Comment (chsienki) [reply src/Compilers/CSharp/Portable/CSharpResources.resx:6569]: F'ing resource files are the bane of my developer existence. (If only there were a better way 👍 )

File: src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs
Comment (mavasani) [reply src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs:46]: So, investigating a bit, the code around this seems to be a bit of a mess.
I thought these special IDE-only ADxxxx IDs and CS and VB ID constants above were removed them long back, but seems like they have lingered around. The comment below seems to indicate the CS and VB IDs above need to be mentioned for live-build deduping: https://github.com/dotnet/roslyn/blob/25f6b6e16dffb08be5cffae4361776e7b6dc71df/src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs#L27-L29

However, that de-duping work is supposed to be handled by the code here:https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Compilers/CSharp/Portable/Compilation/CSharpCompilerDiagnosticAnalyzer.cs#L27.

@jasonmalinowski Can you please file a separate tracking issue for cleaning this up and assign it to me?

**Comment by chsienki** [comment]
Filled https://github.com/dotnet/roslyn/issues/48461 to track improving the generator load coverage on desktop.

**Comment by jasonmalinowski** [comment]
Tagging @dotnet/roslyn-compiler and @dotnet/roslyn-ide for reviews, since this is fixing issues on both sides.

File: src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs
Comment (cston) [inline src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs:253]: >, which these are in desktop [](start = 53, length = 28)

I didn't really understand this part of the comment. Can it be removed? It seems like the previous statement is sufficient.

**Review by cston — APPROVED** [review]
Compiler changes LGTM.

File: src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs
Comment (chsienki) [reply src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs:253]: Ah, it just means that when running the desktop version of the test, the generators we reference are also desktop (because they're in the same assembly). I'll clean it up to make it more clear.

File: src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs
Comment (jasonmalinowski) [reply src/Features/Core/Portable/Diagnostics/AnalyzerHelper.cs:46]: https://github.com/dotnet/roslyn/issues/48448 was filed per @mavasani's request.

**Review by jmarolf — APPROVED** [review]

**Review by tmeschter — COMMENTED** [review]
You're adding localizable resources after the cut-off for 16.8. You should get approval for a tenet exception before merging this in.

File: src/Compilers/VisualBasic/Portable/Errors/MessageProvider.vb
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Errors/MessageProvider.vb:237]: don't love that this is unreachable.  can we still have some sort of test for this (so that if we do add SG support for VB, this fires for .net framework as well)?

**Review by ryzngard — APPROVED** [review]
IDE changes LGTM

**Review by jaredpar — APPROVED** [review]

**Comment by Cosifne** [comment]
@jasonmalinowski  
Is it ready to merge?

**Pull request merged**
